Sources/MapboxMaps/Location/Puck/Puck3D.swift

```diff
@@ -1,6 +1,7 @@
 @_implementationOnly import MapboxCommon_Private
 
 internal final class Puck3D: NSObject, Puck {
+    var showBearingImage: Bool = false
```

Comment:
With this design, we end up with this unused property on Puck3D. I wonder whether we could add the config option to Puck2DConfiguration instead of adding it to LocationOptions. That way it would only exist where it's needed.

Reply:
With the current setting, users can update this value directly with location settings. If move into Puck2DConfiguration, users have to create a new `Puck2D` instance for each update.

Reply:
That's right. That seems okay to me

Reply:
Do we anticipate that developers will be changing this setting frequently? Seems more likely that it gets configured once and never changed.

Reply:
Once users want to change this it would be painful, especially with the custom puck, they will need to rewrite each property with the same value.

Reply:
Here's an example of what it would look like, using the existing 2D-puck-only flag `showsAccuracyRing`:

```
// set initial value
mapView.location.options.puckType = .puck2D(Puck2DConfiguration(topImage: nil, showsAccuracyRing: true))

// update value, preserving other fields
if case .puck2D(var config) = mapView.location.options.puckType {
    config.showsAccuracyRing = false
    mapView.location.options.puckType = .puck2D(config)
}
```

Reply:
That's nice, will move into Puck2DConfiguration